src/services/UniversimeApi/api.ts

```diff
@@ -1,25 +1,31 @@
 import type { ApiResponse } from "@/types/UniversimeApi";
 import * as SwalUtils from "@/utils/sweetalertUtils"
 import axios from "axios";
+import { goTo } from "@/services/routes";
 
 export const api = axios.create({
     baseURL: import.meta.env.VITE_UNIVERSIME_API,
     withCredentials: true,
 });
 
 api.interceptors.response.use(function (response) {
-    if(response.data && response.data.message) {
-        showAlertForResponseData(response.data, false, false)
+    if(response.data) {
+        handleForResponseData(response.data, false, false)
     }
     return response;
 }, function (error) {
-    if(error.response && error.response.data && error.response.data.message) {
-        showAlertForResponseData(error.response.data, true, true)
+    if(error.response && error.response.data) {
+        handleForResponseData(error.response.data, true, true)
     }
     return error.response;
 });
 
-export const showAlertForResponseData = (response: ApiResponse<any>, isModalAsDefault: boolean, isError: boolean) => {
+export const handleForResponseData = (response: ApiResponse<any>, isModalAsDefault: boolean, isError: boolean) => {
+    // handle redirect
+    if(response && response?.redirectTo) {
```

Comment:
Esse if poderia ser apenas `if (response?.redirectTo)` e o `goTo` apenas `goTo(response.redirectTo)`. O `?.` já faz o teste se o `response` existe (não é `null` nem `undefined`) e se passar no teste não precisa checar de novo em baixo.

Não é um erro então não precisaria mudar, mas se quiser deixar o código um pouquinho mais limpo pode trocar.

Reply:
Na verdade, como o método só aceita se for um `ApiResponse`, nem precisaria do `?.` já que tecnicamente nunca seria `null` nem `undefined`.

Reply:
entendi, corrigido